Source/WebGPU/WGSL/GlobalVariableRewriter.cpp

```diff
@@ -1092,7 +1082,38 @@ Vector<unsigned> RewriteGlobalVariables::insertStructs(const UsedResources& used
             auto& global = it->value;
             ASSERT(global.declaration->maybeTypeName());
 
-            entries.append({ binding, &createArgumentBufferEntry(binding, *global.declaration) });
+            entries.append({ metalId, &createArgumentBufferEntry(metalId, *global.declaration) });
+
+            if (m_generatedLayout->bindGroupLayouts.size() <= group)
+                m_generatedLayout->bindGroupLayouts.grow(group + 1);
+
+            BindGroupLayoutEntry entry {
+                .binding = metalId,
```

Comment:
do we need to take into account `global.resource->binding` here or does this not matter based on the specification wording?